src/ngraph/pass/liveness.cpp

```diff
@@ -102,79 +101,31 @@ bool pass::Liveness::run_on_function(shared_ptr<ngraph::Function> function)
 
         for (descriptor::Tensor* tensor_decl : all_tensor_decls)
         {
-            if (!contains(currently_live, tensor_decl))
+            if (currently_live.find(tensor_decl) == currently_live.end())
             {
                 // this is the last node that value is seen in
                 // delete it at the end of the op
                 currently_live.insert(tensor_decl);
-                free_tensor_decls.insert(tensor_decl);
+                if (output_tensors.find(tensor_decl) == output_tensors.end())
```

Comment:
this needs  a comment. Why would we see the def of a tensor before a use (or in the absence of the use)?